data/json/items/gun/308.json

```
@@ -307,7 +316,9 @@
     "type": "GUN",
     "name": { "str_sp": "FN SCAR-H" },
     "description": "A highly accurate and modular battle rifle specially designed for the United States Special Operations Command.  The 'H' in its name stands for heavy, as it uses the powerful .308 round.",
-    "weight": "3640 g",
+    "weight": "3630 g",
+    "volume": "4100 ml",
```

I can't replicate this.
Base model: https://fnamerica.com/products/rifles/fn-scar-17-standard/
Magazine removal: https://imgur.com/a/kERdTj6
SA+Feret: https://imgur.com/a/xmgZz57
Width of 2.75" from your gunivore source, extended OAL from the technical data source. 
```suggestion
    "volume": "5427 ml",
```

---

I'm happy to change this to your suggestion, but I think in this case we are overestimating the volume for the SCAR (and some other rifles).

The SCAR has a built-in extending _and_ folding stock (extension adjustable for the individual shooter).

I guess we first should take the maximum total length of the gun (with the stock fully extended and unfolded) and use that as a base value, then add a folding stock. In this case the weapon has three distinct lengths (and some variable values in between).

The specifications for the FN MK17 LB (the SCAR-H with a 20" long barrel) states the following (https://fnamerica.com/products/rifles/fn-mk-17-lb/):

* 42.0” overall length, with stock fully extended
* 39.5” overall length, with stock fully collapsed
* 32.0” overall length, with stock folded

I've extrapolated this by subtracting four inches off the barrel for the standard 16" barreled version and come up with a total length of 901-965 mm with the stock in the default position (but with adjustable extension), and 711 mm with the stock folded.

Most importantly, the guns are not uniformly wide. Usually the receiver or the front hand guard are the widest points on a gun, whereas most other parts such as the barrel itself is roughly around 1 inch/2.5cm in diameter, and the magazine well, the lower receiver and trigger group are usually slimmer (around 1 - 1.5 inches), so with the method used with estimating the side view area and then multiplying with the stated width I think we may be overestimating the volume a fair bit, at least for assault rifles, carbines, and machine guns.

Just for fun, I loaded a 3D model of a SCAR-L into Blender, removed the magazine, and used the 3D Printing Toolbox to estimate the volume of the model, and it came up with 2498 cm^3. Just saying we might be overestimating a bit here, depending on what we're trying to model with the volume property.